crates/node/builder/src/builder/states.rs

```diff
@@ -220,6 +220,14 @@ impl<T: FullNodeTypes, CB: NodeComponentsBuilder<T>> NodeBuilderWithComponents<T
         launcher.launch_node(self).await
     }
 
+    /// Launches the node with the given closure.
+    pub fn launch_with_fn<L, R>(self, launcher: L) -> R
```

Comment:
what is the difference between this and `launch_with(closure)` given closures now impl `LaunchNode`?

Reply:
this supports arbitrary logic and supports launch functions as closures